### DISCUSSION TITLE
primeng column filter question

### DISCUSSION THREAD

**Discussion by josephplee** [primefaces/community#1828] [PrimeNG] [answered]
Is there a way to set focus on input text filter for menu display for the column filter? 
And is there a way to filter and dismiss said menu display column filter once the user hit the enter button?

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

We've added onShow and onHide emitters to columnFilter, it'll be available in this weeks' release https://github.com/primefaces/primeng/issues/15462

You may achieve this by accessing columnFilter by viewChild and implementing focus logic in onShow.
